src/dll/edbgEcmdDll.C

```diff
@@ -216,108 +162,59 @@ static uint32_t findChipUnitType(const ecmdChipTarget &i_target, uint64_t i_addr
-  else {
-    return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "Unknown chip unit enum:%d\n", i_P9CU);
-  }
-
-  return rc;
-}
-
-//convert chipunit string to enum, as scominfo does not accept strings
-uint32_t p9n_convertCUString_to_enum(std::string cuString, p9ChipUnits_t &o_P9CU) {
-  uint32_t rc = ECMD_SUCCESS;
-
-  if (cuString == "c")          o_P9CU = PU_C_CHIPUNIT;
-  else if (cuString == "eq")    o_P9CU = PU_EQ_CHIPUNIT;
-  else if (cuString == "ex")    o_P9CU = PU_EX_CHIPUNIT;
-  else if (cuString == "xbus")  o_P9CU = PU_XBUS_CHIPUNIT;
-  else if (cuString == "obus")  o_P9CU = PU_OBUS_CHIPUNIT;
-  else if (cuString == "nv")    o_P9CU = PU_NV_CHIPUNIT;
-  else if (cuString == "pec")   o_P9CU = PU_PEC_CHIPUNIT;
-  else if (cuString == "phb")   o_P9CU = PU_PHB_CHIPUNIT;
-  else if (cuString == "mi")    o_P9CU = PU_MI_CHIPUNIT;
-  else if (cuString == "dmi")   o_P9CU = PU_DMI_CHIPUNIT;
-  else if (cuString == "mcs")   o_P9CU = PU_MCS_CHIPUNIT;
-  else if (cuString == "mca")   o_P9CU = PU_MCA_CHIPUNIT;
-  else if (cuString == "mcbist")  o_P9CU = PU_MCBIST_CHIPUNIT;
-  else if (cuString == "perv")  o_P9CU = PU_PERV_CHIPUNIT;
-  else if (cuString == "ppe")   o_P9CU = PU_PPE_CHIPUNIT;
-  else if (cuString == "sbe")   o_P9CU = PU_SBE_CHIPUNIT;
-  else if (cuString == "capp")  o_P9CU = PU_CAPP_CHIPUNIT;
-  else if (cuString == "mc")    o_P9CU = PU_MC_CHIPUNIT;
-  else {
-    return out.error(EDBG_GENERAL_ERROR, FUNCNAME, "Unknown chip unit:%S\n", cuString.c_str());
-  }
+/**
+  * @brief Get chip type
+  *
+  * @param std::string o_chipType - Chip type as output
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+static uint32_t getChipType(std::string &o_chipType) {
+    int rc=ECMD_SUCCESS;
+    uint32_t l_chip_id = 0;
+    uint32_t l_address = 0x100a; //Bits 12:19 tells chipID
+    ecmdDataBuffer l_data;
+    ecmdChipTarget l_target;
+
+    l_target.cageState =  l_target.nodeState = ECMD_TARGET_FIELD_VALID;
+    l_target.slotState = l_target.posState = ECMD_TARGET_FIELD_VALID;
+
+    //Read the 0x100a cfam to get the chip id
+    rc = dllGetCfamRegister(l_target, l_address, l_data);
+    if (rc){
+        return out.error(EDBG_READ_ERROR, FUNCNAME, "cfam read failure!\n");
+    }
 
-  return rc;
+    //bit 12:19 indicates the chip id and based on that unique id we determine
+    //the chip type.
+    l_chip_id = (l_data.getWord(0) & 0x000ff000) >> 12;
+    switch(l_chip_id) {
+        //0x223'd1'049 - p9
+        case 0xd1:
+            o_chipType = CHIP_TYPE_P9;
+            break;
+
+        default:
+            return out.error(EDBG_GENERAL_ERROR,FUNCNAME, "Unrecognised Chip type" PRIx32 "\n", l_chip_id);
+    }
+    return rc;
 }
 
 // Load the device tree and initialise the targets
 static int initTargets(void) {
-  int fd;
-  void *fdt;
-  struct stat stat;
-  static int done = 0;
-
-  if (!done) {
```

Comment:
There were previously issues with calling pdbg_targets_init multiple times, so we had to manage that here.  Does pdbg now take care of that?

Reply:
Is initTargets() called multiple times? i didn't see any issues after removing it. pdbg_targets_init() will give a warning if it is called second time but doesn't crash now.

Reply:
If calling it multiple times is a valid code path, we don't want it throwing a warning at the user for something that is valid.  It looks like now initTargets is only called on dll load, it should be okay.  But if you are seeing the warning in your testing, that need to be fixed.

Reply:
atleast running HW access commands like getcfam, get/putscom, get/putmem, cipinstrucstart/stop all I did not see any issue nor has thrown any warning